jellyfin-rpc/src/lib.rs

```diff
@@ -257,45 +257,49 @@ impl Client {
 
         debug!("Found {} sessions", sessions.len());
 
-        for session in sessions {
-            debug!("Session username is {:?}", session.user_name);
-            if let Some(username) = session.user_name.as_ref() {
-                if self
-                    .usernames
+        self.session = self
+            .usernames
+            .iter()
+            .flat_map(|username| {
+                sessions
                     .iter()
-                    .all(|u| username.to_lowercase() != u.to_lowercase())
-                {
-                    continue;
-                }
-
-                if session.now_playing_item.is_none() {
-                    continue;
-                }
-                debug!("NowPlayingItem exists");
-
-                if session.play_state.is_none() {
-                    continue;
-                }
-                debug!("PlayState exists");
+                    // find any session(s) for the current priority username
+                    .filter(move |session| {
+                        session.user_name.as_ref().is_some_and(|u| u == username)
+                    })
+            })
+            // take the first valid session
+            .find(|session| Self::is_valid_session(session))
+            .map(|session| session.clone().build());
 
-                let session = session.build();
+        Ok(())
+    }
 
-                if session
-                    .now_playing_item
-                    .extra_type
-                    .as_ref()
-                    .is_some_and(|et| et == "ThemeSong")
-                {
-                    debug!("Session is playing a theme song, continuing loop");
-                    continue;
-                }
+    /// Validate that the session we're checking has a now playing item, a play state, and is not a
+    /// theme song.
+    fn is_valid_session(session: &RawSession) -> bool {
```

Comment:
should this validation be done instead as an `impl` of `RawSession` (or `&RawSession`)? Exact logic here might leak application logic into the data model, so I kept it local for now.

This would require a bigger footprint for these changes, but I can see an implementation of `TryFrom` working here for going from `RawSession` to `Session`, yielding `JfResult<Session>`